diff -up cairo-1.2.4/src/cairo-image-surface.c.rgb555 cairo-1.2.4/src/cairo-image-surface.c
--- cairo-1.2.4/src/cairo-image-surface.c.rgb555	2008-01-18 16:45:02.000000000 -0500
+++ cairo-1.2.4/src/cairo-image-surface.c	2008-01-18 16:45:05.000000000 -0500
@@ -141,6 +141,7 @@ _cairo_format_from_pixman_format (pixman
 	break;
     }
 
+    /*
     fprintf (stderr,
 	     "Error: Cairo does not yet support the requested image format:\n"
 	     "\tDepth: %d\n"
@@ -153,6 +154,7 @@ _cairo_format_from_pixman_format (pixman
 	     bpp, am, rm, gm, bm);
 
     ASSERT_NOT_REACHED;
+    */
     return (cairo_format_t) -1;
 }
 
@@ -528,7 +530,7 @@ _cairo_content_from_format (cairo_format
 	return CAIRO_CONTENT_ALPHA;
     }
 
-    ASSERT_NOT_REACHED;
+    /*ASSERT_NOT_REACHED;*/
     return CAIRO_CONTENT_COLOR_ALPHA;
 }
 
